src/tool_cb_hdr.c

```diff
@@ -95,6 +97,44 @@ size_t tool_header_cb(char *ptr, size_t size, size_t nmemb, void *userdata)
     (void)fflush(heads->stream);
   }
 
+  /*
+   * Write etag to file when --etag-save option is given.
+   */
+  if(etag_save->config->etag_save_file && etag_save->stream) {
+    char *r = NULL;
+
+    /* find string with Etag substring */
+    r = strstr(str, "ETag");
```

Comment:
This checks for an `ETag:` response header, right? Then `strstr()`ing for it like this will match too many headers. It should instead rather check for the header to *start* with `ETag:`.

Reply:
yeah you are right, if content of some other header has string ETag that would match to. resolved